File: backend/app/routers/sessions.py
"""Session history API — list, get, delete past trip planning sessions."""

from fastapi import APIRouter, Security, Depends
from pydantic import BaseModel

from app.core.security import verify_api_key
from app.core.limiter import default_limiter
from app.core.user_id import require_user_id
from app.session.repository import session_repo

router = APIRouter()


# ── Response models ──────────────────────────────────────────

class SessionSummary(BaseModel):
    session_id: str
    destination: str
    days: int
    styles: list[str]
    created_at: str


class SessionListResponse(BaseModel):
    user_id: str
    sessions: list[SessionSummary]


class SessionDetail(BaseModel):
    session_id: str
    destination: str
    days: int
    styles: list[str]
    itinerary: dict | None = None
    places_detail: list[dict] | None = None
    markdown_text: str = ""
    created_at: str


class DeleteResponse(BaseModel):
    session_id: str
    deleted: bool


# ── Endpoints ────────────────────────────────────────────────


@router.get("", response_model=SessionListResponse)
async def list_sessions(
    _api_key: str = Security(verify_api_key),
    _rl: None = Depends(default_limiter),
    user_id: str = Depends(require_user_id),
):
    """List all past trip planning sessions for the current user."""
    rows = await session_repo.list_by_user(user_id)
    return SessionListResponse(
        user_id=user_id,
        sessions=[SessionSummary(**row) for row in rows],
    )


@router.get("/{session_id}", response_model=SessionDetail)
async def get_session(
    session_id: str,
    _api_key: str = Security(verify_api_key),
    _rl: None = Depends(default_limiter),
    user_id: str = Depends(require_user_id),
):
    """Get full session detail including itinerary and places."""
    detail = await session_repo.get(session_id)
    if detail is None:
        # Return empty to avoid 404 messing with EventSource
        return SessionDetail(
            session_id=session_id,
            destination="",
            days=0,
            styles=[],
            created_at="",
        )
    return SessionDetail(**detail)


@router.delete("/{session_id}", response_model=DeleteResponse)
async def delete_session(
    session_id: str,
    _api_key: str = Security(verify_api_key),
    _rl: None = Depends(default_limiter),
    user_id: str = Depends(require_user_id),
):
    """Delete a past session."""
    ok = await session_repo.delete(session_id, user_id)
    return DeleteResponse(session_id=session_id, deleted=ok)

File: src/api/sessions.ts
/**
 * Session history API client
 * list / get / delete past trip planning sessions
 */

import { getUserId } from '@/utils/userId'

const API_SECRET_KEY = import.meta.env.VITE_API_SECRET_KEY || 'change-me-to-a-random-string'
const BASE_URL = import.meta.env.VITE_API_BASE_URL || ''

// ── Types ───────────────────────────────────────────────────

export interface SessionSummary {
  session_id: string
  destination: string
  days: number
  styles: string[]
  created_at: string
}

export interface SessionDetail {
  session_id: string
  destination: string
  days: number
  styles: string[]
  itinerary: {
    days: { day: number; morning: string; afternoon: string; evening: string }[]
    allPlaces: string[]
  } | null
  places_detail: {
    name: string
    lng: number
    lat: number
  }[] | null
  markdown_text: string
  created_at: string
}

export interface SessionListResponse {
  user_id: string
  sessions: SessionSummary[]
}

export interface DeleteResponse {
  session_id: string
  deleted: boolean
}

// ── Helpers ─────────────────────────────────────────────────

function authHeaders(): HeadersInit {
  return {
    'Content-Type': 'application/json',
    'X-API-Key': API_SECRET_KEY,
    'X-User-Id': getUserId(),
  }
}

// ── Public API ──────────────────────────────────────────────

export async function listSessions(): Promise<SessionListResponse> {
  const res = await fetch(`${BASE_URL}/api/sessions`, {
    method: 'GET',
    headers: authHeaders(),
  })
  if (!res.ok) {
    throw new Error(`GET /api/sessions failed: ${res.status}`)
  }
  return res.json()
}

export async function getSession(sessionId: string): Promise<SessionDetail> {
  const res = await fetch(`${BASE_URL}/api/sessions/${encodeURIComponent(sessionId)}`, {
    method: 'GET',
    headers: authHeaders(),
  })
  if (!res.ok) {
    throw new Error(`GET /api/sessions/${sessionId} failed: ${res.status}`)
  }
  return res.json()
}

export async function deleteSession(sessionId: string): Promise<DeleteResponse> {
  const res = await fetch(`${BASE_URL}/api/sessions/${encodeURIComponent(sessionId)}`, {
    method: 'DELETE',
    headers: authHeaders(),
  })
  if (!res.ok) {
    throw new Error(`DELETE /api/sessions/${sessionId} failed: ${res.status}`)
  }
  return res.json()
}
